src/main/java/gov/va/viers/cdi/emis/ws/endpoint/MilitaryInfoEndpoint.java

```diff
@@ -30,17 +42,129 @@
       localPart = "eMISmilitaryServiceEligibilityRequest")
   @ResponsePayload
   public JAXBElement<EMISmilitaryServiceEligibilityResponseType> getServiceEligibility(
-      @RequestPayload InputEdiPiOrIcn request) {
-    /*     This should probably be wrapped in some null checks, not sure what EMIS does in those cases
+      @RequestPayload InputEdiPiOrIcn request,
+      @SoapHeader(value = "{http://viers.va.gov/cdi/CDI/commonService/v2}inputHeaderInfo")
+          SoapHeaderElement soapHeaderElement) {
+
+    InputHeaderInfo requestSoapHeaders = new InputHeaderInfo();
+    requestSoapHeaders = getInputHeaderInfo(soapHeaderElement, requestSoapHeaders);
+
+    try {
+      if (!request.getEdipiORicn().getInputType().equals("EDIPI")) {
+        return generateMissingOrInvalidEdipiEssError(
+            requestSoapHeaders, "MIS-ERR-03", "INVALID_IDENTIFIER");
+      } else if (request.getEdipiORicn().getEdipiORicnValue().isEmpty()) {
+        return generateMissingOrInvalidEdipiEssError(
+            requestSoapHeaders, "MIS-ERR-02", "MISSING_EDIPI");
+      }
+    } catch (DatatypeConfigurationException e) {
+      LOGGER.error("error while generating essError", e);
+    }
+
+    /* This should probably be wrapped in some null checks, not sure what EMIS does in those cases
     though with regards to returning that there was bad input*/
     JAXBElement<gov.va.schema.emis.vdrdodadapter.v2.EMISmilitaryServiceEligibilityResponseType>
         dodResponse =
             dodClient.getMilitaryServiceEligibilityResponse(
                 request.getEdipiORicn().getEdipiORicnValue());
 
+    try {
+      if (dodResponse.getValue().getESSError() != null) {
+        if (dodResponse.getValue().getESSError().getESSResponseCode().equals("ERROR")
+            && dodResponse.getValue().getESSError().getEssText().equals("INVALID_EDIPI_INPUT")) {
+          EMISMapperImpl emisMapper = new EMISMapperImpl();
+          ESSErrorType essErrorType =
+              emisMapper.mapESSErrorType(dodResponse.getValue().getESSError());
+          return generateBadFormatEdipiEssError(essErrorType, requestSoapHeaders);
+        }
+      }
+    } catch (DatatypeConfigurationException e) {
+      LOGGER.error("error when generating badFormatEdipiEssError", e);
+    }
+
     EMISmilitaryServiceEligibilityResponseType noJaxbResponse;
-    noJaxbResponse = EMISMapper.INSTANCE.mapEMISmilitaryServiceEligibilityResponseType(dodResponse.getValue());
+    noJaxbResponse =
+        EMISMapper.INSTANCE.mapEMISmilitaryServiceEligibilityResponseType(dodResponse.getValue());
 
     return objectFactory.createEMISmilitaryServiceEligibilityResponse(noJaxbResponse);
   }
+
+  private InputHeaderInfo getInputHeaderInfo(
+      SoapHeaderElement soapHeaderElement, InputHeaderInfo requestSoapHeaders) {
```

Comment:
Passing in requestSoapHeaders is not necessary. Create the object in the method instead.